examples/apps/ai-collab/src/components/TaskGroup.tsx

```diff
@@ -245,6 +337,34 @@ export function TaskGroup(props: {
 					></TaskGroupDiffModal>
 				)}
 
+				{undoStack.length > 0 && (
+					<Button
+						variant="contained"
+						color="error"
+						onClick={() => {
+							const revertible = undoStack[undoStack.length - 1];
+							revertible?.revert();
+							undoStack.pop();
+						}}
+					>
+						Undo
+					</Button>
+				)}
+
+				{redoStack.length > 0 && (
+					<Button
+						variant="contained"
+						color="info"
+						onClick={() => {
+							const revertible = redoStack[redoStack.length - 1];
+							revertible?.revert();
+							redoStack.pop();
```

Comment:
First I was expecting `.pop()` to be called first, since it returns the element that was removed so we could do `const revertible = redoStack.pop()`. But then I thought that if the `revert()` fails, we probably want to keep that revertible in the stack, so if we start by `pop()`ing it we would have to catch errors and put it back. So I like this order of operations that doesn't need a try-catch, but I'd add a comment to make it clear that's why we're doing it this way. Same for the undo stack code above.